ODIN_II/SRC/ast_elaborate.cpp

```diff
@@ -976,6 +976,24 @@ ast_node_t* build_hierarchy(ast_node_t* node, ast_node_t* parent, int index, sc_
                     error_message(AST, node->loc,
                                   "%s", "Cannot perform operation with nonexistent value");
                 }
+                // TODO convert to bufifs
```

Comment:
buffs are gate instantiation nodes and as such they expect their output to be a child, 
I don't think here is the right place to do this unless we want to do lookahead in the tree or unless this optimization is needed at the AST level for inouts.

@djns99 I don't recollect you needing that information this early on, if you feel it would help I'll work on this, otherwise, I might just remove it and do the recognition during the netlist creation phase.

Reply:
I don't do any checks that drivers are Z so I don't need to know about this anywhere I believe. As long as the simulator resolves the gate to a Z it should be fine.

Reply:
just to clarify, I was wondering if you needed to know about the bufifs early on or not.

Reply:
No inouts and multidriver do not care about any specific components. They only care about the simulator outputting the correct value